source/architecture.rst

```diff
@@ -1,230 +1,245 @@
+In Kafka, a specialized topics structure is used to insure isolation between
+different users and applications data, enabling a multi-tenant infrastructure.
 
-The subscription manager service makes use of an in-memory database for efficiency. It adds context to Apache Kafka,
-making it possible that internal or even external services are able to subscribe or query data based on context. The
-subscription manager is also a distributed service to avoid it being a single point of failure or even a bottleneck for
-the architecture.
+The flow-broker service makes use of an in-memory database for efficiency. It
+adds context to Apache Kafka, making it possible that internal or even external
+services are able to subscribe or query data based on context. Flow-broker is
+also a distributed service to avoid it being a single point of failure or even
+a bottleneck for the architecture.
 
-To keep a certain level of compatibility with Fiware services, meaning, for using Fiware services and components in the
-*dojot* platform with the minimal amount of adaptations, we added a NGSI interface translation service.
+To keep a certain level of compatibility with NGSI-compatible components, we
+added a NGSI interface translation service.
 
-Device Manager
+DeviceManager
 **************
 
-The IoT Device manager is a core entity, responsible for maintaining the devices data models and its abstractions, it is
-also responsible for propagating this models to services that are interested in this kind of information, for example,
-the iot-agent.
+DeviceManager is a core entity which is responsible for keeping device and
+templates data models. It is also responsible for publishing any updates to all
+interested components (namely IoT agents, history and subscription manager)
+through Kafka.
 
-This service is stateless, having its data persisted to a database, with data isolation for users and applications,
-making possible a multi-tenant architecture for the middleware.
+This service is stateless, having its data persisted to a database, with data
+isolation for users and applications, making possible a multi-tenant
+architecture for the middleware.
 
-Iot-agent
+IoT Agent
 *********
 
-The iot-agent is an adaptation service between the data model of the context broker and the devices data models. The
-*dojot* platform can have multiple iot-agents, each one of them being specialized in a specific protocol like, for
-instance, MQTT/JSON, CoAP/LWM2M and HTTP/JSON.
+An IoT agent is an adaptation service between physical devices and *dojot* core
+components. It could be understood as a *device driver* for a set of devices.
+The *dojot* platform can have multiple iot-agents, each one of them being
+specialized in a specific protocol like, for instance, MQTT/JSON, CoAP/LWM2M
+and HTTP/JSON.
+
+It is also responsible for all security mechanisms used in communication
+channels between the devices it deals with and the component itself.
```

Comment:
This phrase is odd.

I think something like what is shown below would sound better

"It is also responsible for the mechanisms to ensure that the communication channel with the devices it handles is secure."

Reply:
How about:
```
It is also responsible to ensure that it communicates with devices using secure channels.
```